Tests/Realm.Tests/Database/NotificationTests.cs

```diff
@@ -1627,7 +1627,7 @@ public void DictionaryOfPrimitives_SubscribeForNotifications_ShallowHasNoEffect(
                 dict.Clear();
             });
 
-            VerifyNotifications(changesets, expectedDeleted: new[] { 0 }, expectedCleared: false);
+            VerifyNotifications(changesets, expectedDeleted: new[] { 0 }, expectedCleared: true);
```

Comment:
The behaviour of dicts and sets has been aligned with lists in regard to `Clear`